packages/headless/src/controllers/core/facets/facet/headless-core-facet-options.ts

```diff
@@ -89,14 +91,63 @@ export interface FacetSearchOptions {
   query?: string;
 }
 
-export const facetOptionsSchema = new Schema<Required<FacetOptions>>({
+export const facetOptionsSchema = new Schema({
   facetId,
   field,
   // TODO: Remove on next major version
   delimitingCharacter,
   filterFacetCount,
   injectionDepth,
   numberOfValues,
-  sortCriteria: new StringValue({constrainTo: facetSortCriteria}),
   facetSearch,
 });
+
+export const facetOptionSortCriteriaStringSchema = new Schema({
+  sortCriteria: new StringValue({
+    constrainTo: facetSortCriteria,
+  }),
+});
+
+export const facetOptionSortCriteriaExplicitSchema = new Schema({
+  sortCriteria: new RecordValue({
+    values: {
+      type: new StringValue({
+        constrainTo: [...facetSortCriteria, 'custom'],
+        required: true,
+        emptyAllowed: false,
+      }),
+      customSort: new ArrayValue({
+        required: false,
+        each: new StringValue({emptyAllowed: false}),
+      }),
+    },
+  }),
+});
+
+export function validateFacetOptions(
+  engine: CoreEngine,
+  options: Required<FacetOptions>,
+  schema: Schema<object>,
+  functionName: string
+) {
+  const {sortCriteria, ...allOthers} = options;
+  validateOptions(engine, schema, allOthers, functionName);
+
+  if (!sortCriteria) {
+    return;
+  }
+
+  const schemaForSortCriteria = getSortCriteriaPolymorphicSchema(sortCriteria);
+
+  validateOptions(engine, schemaForSortCriteria!, {sortCriteria}, functionName);
+}
+
+export function getSortCriteriaPolymorphicSchema(
+  sortCriteria: FaceSortCriterionStringOrExplicit
+) {
+  if (typeof sortCriteria === 'string') {
+    return facetOptionSortCriteriaStringSchema;
+  }
+
+  return facetOptionSortCriteriaExplicitSchema;
```

Comment:
Does this mean that we can either custom sort or do a full sort, but not both?

From the JIRA on the search API, they seem to imply that the purpose of the feature is to be able to sort the top values (pin them) and still see more values below those. I would expect that we could (eventually) specify both properties independently.

Reply:
You can see the doc here: 

https://docs.coveo.com/en/13/api-reference/search-api#tag/Search-V2/operation/searchUsingPost

Facets -> Sort criteria -> Click on the `one of` small button.